main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala

```diff
@@ -319,6 +320,35 @@ object CodeActionProvider {
     command = None
   )
 
+  /**
+    * Returns a list of quickfix code action to suggest possibly correct spellings.
+    * Uses Levenshtein Distance to find closed spellings.
+    */
+  private def mkFixMisspelling(qn: Name.QName, loc: SourceLocation, env: Map[String, Symbol.VarSym], uri: String): List[CodeAction] = {
+    if (qn.ident.name.length > 3) {
+      val possibleNames = env.toList.map(_._1).filter(n => (n.length - qn.ident.name.length).abs < 3)
```

Comment:
Does Scala not have a fillterKeys or something? Maybe not?

Reply:
Otherwise a fold might be better? or even an imperative loop buidning up a ListBuffer of CodeActions.

Reply:
Do you mean something like this to avoid intermediate List construction?

```scala
if (qn.ident.name.length > minLength) 
      env.toList.foldLeft(List.empty[CodeAction]) { (acc, kv) =>
        val name = kv._1
        if ((name.length - qn.ident.name.length).abs < minDistance && Similarity.levenshtein(qn.ident.name, name) < minDistance)
          mkCorrectSpelling(name, loc, uri) :: acc
        else 
          acc
      }
    else 
      Nil
```

Reply:
Hmm. It was not so much about performance. It was about the clarify of the code, but perhaps what you have is clearer than a fold...

Reply:
I will let you choose.

Reply:
I would prefer the older one. I have modified it a bit to make it better looking :)